splito/lohi/_lo.py

```diff
@@ -0,0 +1,157 @@
+from rdkit import DataStructs
+import numpy as np
+from tqdm import tqdm
+import datamol as dm
+import functools
+from loguru import logger
+
+
+class LoSplitter:
+    def __init__(self, threshold=0.4, min_cluster_size=5, max_clusters=50, std_threshold=0.60):
+        """
+        Creates the splitter object.
+
+        Args:
+            threshold: molecules with a similarity higher than this value are considered similar.
+            min_cluster_size: number of molecules per cluster.
+            max_clusters: maximum number of selected clusters. The remaining molecules go to the training set.
+            std_threshold: Lower bound of the acceptable standard deviation for a cluster. It should be greater than measurement noise.
+                            If you're using ChEMBL-like data, set it to 0.60 for logKi and 0.70 for logIC50.
+                            Set it lower if you have a high-quality dataset.
+        """
+        self.threshold = threshold
+        self.min_cluster_size = min_cluster_size
+        self.max_clusters = max_clusters
+        self.std_threshold = std_threshold
+
+    def split(self, smiles, values, n_jobs=-1, verbose=1):
+        """
+        Split the dataset of smiles and their continuous values.
+
+        Args:
+            smiles: list of smiles.
+            values: list of their continuous activity values.
+            verbose: set to 0 to turn off progressbar.
+
+        Returns:
+            train_idx: list of train indices.
+            clusters_idx: list of lists of cluster indices.
+        """
+        if not isinstance(smiles, np.ndarray):
+            smiles = np.array(smiles)
+        if not isinstance(values, np.ndarray):
+            values = np.array(values)
+
+        train_idx, clusters_idx = self._select_distinct_clusters(smiles, values, n_jobs, verbose)
+        train_idx = list(train_idx)
+
+        # Move one molecule from each cluster to the training set
```

Comment:
Here the last compound in cluster is moved to train. So are the molecules in one cluster sorted?
It's often the case in LO, the compounds in earlier cycles( which are considered as training set here) have relatively lower activities and people aim to discovery molecules with higher potencu in later cycle (considered as test set).
This can be taken into account if it has been so far.

Reply:
Sorry for that, it's just me being unpythonic :)

Cluster molecules are not sorted, but I needed a way to mark the centroid of each cluster, so I decided to put it at the end of the lists. That was a quick solution, but since it confused you, I'll make returning the centroid more explicit now.